src/v0/destinations/discord/util.js

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable no-nested-ternary */
+const { getFieldValueFromMessage } = require("../../util");
+
+/**
+ * get the name value from following heirarchy
+ * traits.name
+ * traits.firstName+traits.lastName (or either)
+ * traits.username
+ * traits.email
+ * properties.email
+ * userId
+ * anonymousId
+ * @param {*} message
+ * @returns uname
+ */
+const getName = message => {
+  const traits = getFieldValueFromMessage(message, "traits");
+  let uName;
+  if (traits) {
+    uName =
+      traits.name ||
+      (traits.firstName
+        ? traits.lastName
+          ? `${traits.firstName}${traits.lastName}`
+          : traits.firstName
+        : undefined) ||
+      traits.username ||
+      (message.properties ? message.properties.email : undefined) ||
+      traits.email ||
+      (message.userId ? `User ${message.userId}` : undefined) ||
+      `Anonymous user ${message.anonymousId}`;
+  } else {
+    uName =
+      (message.properties ? message.properties.email : undefined) ||
+      (message.userId ? `User ${message.userId}` : undefined) ||
+      `Anonymous user ${message.anonymousId}`;
+  }
```

Comment:
If the traits object is not there it won't work. That's why it is implemented in this manner.

Reply:
We can use optional chaining to test the availability of traits. Something as following:
```
 uName =
      traits?.name ||
      (traits?.firstName
        ? traits?.lastName
          ? `${traits.firstName}${traits.lastName}`
          : traits.firstName
        : undefined) ||
      traits?.username ||
      (message.properties ? message.properties.email : undefined) ||
      traits?.email ||
      (message.userId ? `User ${message.userId}` : undefined) ||
      `Anonymous user ${message.anonymousId}`;
```

Reply:
> We should avoid reusing utility functions from another destination, and rather accomadate any such functionalities in root-level utility functions. There can be unnecessary issues in a destination, if going ahead, some change with respect to the original destination is made in the utility function and it breaks something at the other destination.
> 
> Non blocking comment

We have used the utils file mainly and since both of them are identical destinations we are using the same transformation plan and so if there is a change in one it should reflect in the second as well to keep parity on how we support similar types of destinations.
We can keep it in global utils as well would it makes sense as only two destinations will be using it ?